en/resources/tutorials/codablerouting.md

```diff
@@ -0,0 +1,172 @@
+    public var title: String       
+    public var order: Int      
+    public var completed: Bool 
+    public var id: Int?         // Additional field for identifier
+}
+```
+
+And the storeHandler function would then set it:
+
+```swift
+func storeHandler(todo: ToDo, completion: (ToDo?, RequestError?) -> Void ) -> Void {
+    var todo = todo
+    let id = todoStore.count
+    todo.id = id
+    todoStore.append(todo)
+    completion(todo, nil)
+} 
+```
+
+### 2. Responding with an additional Identifier value
+Alternatively you can choose to respond with an additional value in the completion handler that conforms to the Identifier protocol, which String and Int have been extended to conform to already. The additional identifier value is then written into the “Location” header of the response.
+
+Here the storeHandler would become the following:
+```swift
+func storeHandler(todo: ToDo, completion: (Int?, ToDo?, RequestError?) -> Void ) -> Void {
+    var todo = todo
+    let id = todoStore.count
+    todoStore.append(todo)
+    completion(id, todo, nil)
+}
+```
+
+## Building a Read API with a Codable Route for GET requests
+
+Now that the Kitura application can store a ToDo item and inform the client where it is, there needs to be a REST API to allow the client to read it. This is done by registering a handler for GET requests on ‘/todos/<id>’.
+
+```swift
+router.get("/todos", handler: getOneHandler)
+```
+
+The registration of the handler is only against the “/todos” path, but we also need to accept an <id> sent from the client. This is done in the handler itself:
+
+```swift
+func getOneHandler(id: Int, completion: (ToDo?, RequestError?) -> Void ) -> Void {
+    completion(todoStore[id], nil)
+}
+```
+
+Here the Kitura router itself parses the URI path, and converts the id into an Int before calling the handler. 
+Similar to the way that you can specify incoming data parameters as a type that implements Codable, it is possible to specify identifier parameters as any type that implements Identifier – which is a protocol provided by Kitura.
+
+## Building Identifiers with the Identifier Protocol
+Defining an Identifier for use with Codable Routing has two requirements. The first is that it must implement the Identifier protocol, creating an instance that can be used as an identifier from a string constructor.
+
+The following is an example of a custom Identifier called Item:
+```swift 
+public struct Item: Identifier {
+    public var value: String
+    public let id: Int
+ 
+    public init(value: String) throws {
+        if let id = Int(value) {
+            self.id = id
+            self.value = value
+        } else {
+            throw IdentifierError.invalidValue
+        }
+    }
+}
+```
+
+The second requirement is that the original String used in the constructor must be stored in the value field. The reason for this is to allow the Identifier to be used when data types and APIs, with the KituraKit client.
```

Comment:
The latter half of this sentence doesn't make sense. `The reason for this is to allow the Identifier to be used when data types and APIs... what?`